### PR TITLE
kvutils: Make `Step` a trait to carry the logging context more easily.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
Implicits and lambdas don't work well together.

I don't know if this is a good idea. I find the `Step` definitions a little easier to read, and I appreciate avoid passing `loggingContext` around explicitly (although this mostly happens in tests). That said, there's undeniably more code, and more nesting, which you might find less pleasant.

Let me know what you think.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by fabiotudone-da — APPROVED** [review]
I prefer it like this.

**Comment by SamirTalwar-DA** [comment]
Gonna wait for one more approval, just in case someone has misgivings.

**Comment by SamirTalwar-DA** [comment]
I've waited long enough.

**Pull request merged**
